### ISSUE TITLE
ActiveCampaign - Not all fields are fetched

### ISSUE THREAD

**Issue by geertvansoest** [solspace/craft3-freeform#1197]
**Describe the bug or issue you're experiencing**
In Freeform 5, when you fetch the fields from your ActiveCampaign integration, just a maximum of 25 fields are visible. This are the default fields (listId, firstName, lastName, email, phone) and only 20 fields from the `/fields` endpoint.

**Steps to reproduce**
1. In your Craft admin panel, go the a form and open the "Integrations" tab.
2. At the left side navigation, select your ActiveCampaign integration.
3. At the "Contact mapping" field, click the icon to load the fields.

**Expected behavior**
To load _all_ fields from your ActiveCampaign integration.

**Plugin Info:**
Freeform version: 5.1.8

**Additional context**
Freeform uses the `/fields` endpoint to load the fields from ActiveCampaign (https://developers.activecampaign.com/reference/retrieve-fields). If no limit is provided, this endpoint defaults to only 20 fields. 

The most easy fix could be to simply add a very high limit. I think this could be a (temporary) fix to offer a solution for all current implementations as soon as possible:
```
$response = $client->get($this->getEndpoint('/fields?limit=9999'));
```
At https://github.com/solspace/craft-freeform/blob/v5/packages/plugin/src/Integrations/CRM/ActiveCampaign/BaseActiveCampaignIntegration.php#L234

An alternative could be to do more requests if the `total` in the API response is higher then the provided `limit`, using the `offset` property (https://developers.activecampaign.com/reference/pagination):
```
"meta": {
   "total": "150",
   "selected": null
}
```

The same issue may occur in `fetchDealFields()` and `fetchAccountFields()`, but I'm not sure since we don't use these in our project.

**Comment by geertvansoest** [comment]
For now, we created this local patch file in our project to fix this issue:
```
--- /vendor/solspace/craft-freeform/packages/plugin/src/Integrations/CRM/ActiveCampaign/BaseActiveCampaignIntegration.php	2024-04-11 11:39:37
+++ /vendor/solspace/craft-freeform/packages/plugin/src/Integrations/CRM/ActiveCampaign/BaseActiveCampaignIntegration.php	2024-04-11 11:39:37
@@ -231,7 +231,7 @@
     private function fetchContactFields(Client $client, string $category): array
     {
         try {
-            $response = $client->get($this->getEndpoint('/fields'));
+            $response = $client->get($this->getEndpoint('/fields?limit=9999'));
         } catch (\Exception $exception) {
             $this->processException($exception, self::LOG_CATEGORY);
         }
```

At the [documentation](https://developers.activecampaign.com/reference/pagination), it says the maximum limit is 100, but we receive all our fields (150) using this fix.

**Comment by kjmartens** [comment]
Thanks for this @geertvansoest. We'll have this fixed in Freeform 5.1.11 (available shortly). 🙂
